### PR TITLE
Primary scroll view support for SingleChildScrollView

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/single_child_scroll_view.dart [hunk 1]
```diff
@@ -42,11 +43,16 @@ class SingleChildScrollView extends StatelessWidget {
...
+    this.primary: false,
     this.physics,
     this.controller,
     this.child,
   }) : super(key: key) {
     assert(scrollDirection != null);
+    assert(primary != null);
+    assert(controller == null || !primary,
+       'Primary ScrollViews obtain their ScrollController via inheritance from a PrimaryScrollController widget. '
+       'You cannot both set primary to true and pass an explicit controller.');
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/single_child_scroll_view.dart]: nit: missing newline before )

Comment (Hixie) [inline packages/flutter/lib/src/widgets/single_child_scroll_view.dart]: looks like i screwed that up in scroll_view.dart too

**Comment by Hixie** [comment]
LGTM if you add a test

**Comment by cbracken** [comment]
Added!

File: packages/flutter/lib/src/widgets/single_child_scroll_view.dart [hunk 2]
```diff
@@ -42,11 +43,17 @@ class SingleChildScrollView extends StatelessWidget {
     this.scrollDirection: Axis.vertical,
     this.reverse: false,
     this.padding,
+    this.primary: false,
```
Comment (tvolkert) [inline packages/flutter/lib/src/widgets/single_child_scroll_view.dart]: Side-note: our widget constructors need dartdoc (I was gonna say this param should be documented, only to realize the whole constructor needs to be documented :-))

Comment (abarth) [inline packages/flutter/lib/src/widgets/single_child_scroll_view.dart]: Yes, the scrolling classes are missing many dartdocs.  It's on my list for this week.

**Pull request merged**